internal/contentmanager/contentmanager.go

```diff
@@ -0,0 +1,203 @@
+			u.SetGroupVersionKind(gvk)
+			ul := &unstructured.UnstructuredList{}
+			ul.SetGroupVersionKind(gvk.GroupVersion().WithKind(listKind))
+
+			scheme.AddKnownTypeWithName(gvk, u)
+			scheme.AddKnownTypeWithName(gvk.GroupVersion().WithKind(listKind), ul)
+			// Adding the common meta schemas to the scheme for the GroupVersion
+			// is necessary to ensure the scheme is aware of the different operations
+			// that can be performed against the resources in this GroupVersion
+			metav1.AddToGroupVersion(scheme, gvk.GroupVersion())
+		}
+	}
+
+	return scheme, nil
+}
+
+// ManageContent configures a controller-runtime cache.Cache and establishes watches for the provided resources.
+// It utilizes the provided ClusterExtension to set a DefaultLabelSelector on the cache.Cache
+// to ensure it is only caching and reacting to content that belongs to the ClusterExtension.
+// For each client.Object provided, a new source.Kind is created and used in a call to the Watch() method
+// of the provided controller.Controller to establish new watches for the managed resources.
+func (i *instance) ManageContent(ctx context.Context, ctrl controller.Controller, ce *v1alpha1.ClusterExtension, objs []client.Object) error {
+	// Return nil if the objs slice is empty before any further action
+	if len(objs) == 0 || ce == nil || ctrl == nil {
+		return nil
+	}
+
+	cfg, err := i.rcm(ctx, ce, i.baseCfg)
+	if err != nil {
+		return fmt.Errorf("getting rest.Config for ClusterExtension %q: %w", ce.Name, err)
+	}
+
+	scheme, err := buildScheme(objs)
+	if err != nil {
+		return fmt.Errorf(
+			"building scheme for %s; %w",
+			ce.GetName(),
+			err,
+		)
+	}
+
+	tgtLabels := labels.Set{
+		oclabels.OwnerKindKey: v1alpha1.ClusterExtensionKind,
+		oclabels.OwnerNameKey: ce.GetName(),
+	}
+
+	c, err := cache.New(cfg, cache.Options{
+		Scheme:               scheme,
+		DefaultLabelSelector: tgtLabels.AsSelector(),
+	})
+	if err != nil {
+		return fmt.Errorf("creating cache for ClusterExtension %q: %w", ce.Name, err)
+	}
+
+	for _, obj := range objs {
+		err = ctrl.Watch(
+			source.Kind(
+				c,
+				obj,
+				handler.TypedEnqueueRequestForOwner[client.Object](
+					scheme,
+					i.mapper,
+					ce,
+				),
+			),
+		)
+		if err != nil {
+			return fmt.Errorf("creating watch for ClusterExtension %q managed resource %s: %w", ce.Name, obj.GetObjectKind().GroupVersionKind(), err)
+		}
+	}
+
+	// TODO: Instead of stopping the existing cache and replacing it every time
+	// we should stop the informers that are no longer required
+	// and create any new ones as necessary. To keep the initial pass
+	// simple, we are going to keep this as is and optimize in a follow-up.
+	// Doing this in a follow-up gives us the opportunity to verify that this functions
+	// as expected when wired up in the ClusterExtension reconciler before going too deep
+	// in optimizations.
+	if data, ok := i.extensionCaches[ce.GetName()]; ok {
+		data.Cancel()
+	}
+
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
See https://github.com/operator-framework/operator-controller/pull/1001#discussion_r1672182549

Reply:
Based on my last comment up there, I'd suggest:
1. Name the new `ctx` something different (e.g. `cacheCtx`) to avoid shadowing the original `ctx`
2. Use `context.Background()` as the base context, not `ctx`.